tests/tests_unit/test_unit.py

```diff
@@ -294,16 +294,124 @@ def test_get_values_from_bids_formatted_name_error(self):
                 "sub",
                 return_as_int=True,
             )
-        assert "Invalid character in sub number: sub-a_date-12345" == str(
+
+        assert "Invalid character in subject or session value: a" == str(
             e.value
         )
 
-    # -------------------------------------------------------
-    #
-    #
-    # ---------------
-    # Utlis
-    # ----------------------------------------------------------------------
+    # Test getting max sub or ses num from list
+    # ----------------------------------------------------------------------------------
+
+    @pytest.mark.parametrize("prefix", ["sub", "ses"])
+    @pytest.mark.parametrize("default_num_value_digits", [0, 1, 11, 99, 101])
+    def test_get_max_sub_or_ses_num_and_value_length_empty(
+        self, prefix, default_num_value_digits
+    ):
+        """
+        When the list of sub or ses names is empty, the returned max number
+        should be zero and the `default_num_value_digits` be set to the passed default
+        """
+        (
+            max_value,
+            num_digits,
+        ) = folders.get_max_sub_or_ses_num_and_value_length(
+            [], prefix, default_num_value_digits
+        )
+
+        assert max_value == 0
+        assert num_digits == default_num_value_digits
+
+    @pytest.mark.parametrize("prefix", ["sub", "ses"])
+    def test_get_max_sub_or_ses_num_and_value_length_error(self, prefix):
+        """
+        An error will be shown if the sub or ses value digits are inconsistent,
+        because it is not possible to return the number of values required.
+
+        A warning should be shown in that the number of value digits are
+        inconsistent, as the user may be confused as to the real max.
+        """
+        bad_num_values_names = [
+            f"{prefix}-001",
+            f"{prefix}-02",
+            f"{prefix}-003",
+        ]
+
+        with pytest.raises(BaseException) as e:
```

Comment:
you can use the "match" argument of `pytest.raises` to search for the expected error message.
But perhaps you want to do that across all tests in a future PR. It makes no actual difference, but it will save you some code lines.

Reply:
Thanks it is a v cool argument I tried this in #242 but unfortunately it has some funny behaviour with regexp that made it more trouble that it's worth in few cases, unless I overlooked something..